afl-fuzz.c

```diff
@@ -1332,29 +1320,34 @@ static void cull_queue(void) {
 
   while (q) {
     q->favored = 0;
+    q->rand = UR(INT_MAX);
     q = q->next;
   }
 
-  /* Let's see if anything in the bitmap isn't captured in temp_v.
-     If yes, and if it has a top_rated[] contender, let's use it. */
-
-  for (i = 0; i < MAP_SIZE; i++)
-    if (top_rated[i] && (temp_v[i >> 3] & (1 << (i & 7)))) {
-
-      u32 j = MAP_SIZE >> 3;
-
-      /* Remove all bits belonging to the current entry from temp_v. */
-
-      while (j--) 
-        if (top_rated[i]->trace_mini[j])
-          temp_v[j] &= ~top_rated[i]->trace_mini[j];
+   for (i = 0; i < MAP_SIZE; i++) {
+    if (potential_favored_list[i]) {
+      struct potential_favored_input* potential_input = potential_favored_list[i];
+      struct queue_entry* new_top_rated;
+      int minimum_random_number = INT_MAX;
+      while (potential_input) {
```

Comment:
Add weighted version of this with weight 1 (for now).

Reply:
```
w := 1.0
rid := MaxUint64
while (; 1.0 <= w; w -= 1.0) {
	r := rndUint64()
	if r < rid {
		rid = r
	}
}
if (0.0 < w && rndFloatBetweenZeroAndOne() < w) {
	r := rndUint64()
	if r < rid {
		rid = r
	}
}
```

Reply:
Yeah, that condition is meant to only be executed if the weight is not an integer; for example, if `w` is initialized to 2.3 it should perform two iterations of the loop and then enter the conditional with 30% probability. Does that make sense?

Reply:
@wuestholz, ah I got it. I updated the code now. So basically a new variable `rid` would be used instead of this one? 
```
 int minimum_random_number = INT_MAX;
```

Reply:
@jiradeto Thanks! Yeah, the name doesn't matter so much. I just used the short name in the pseudo code.